addons/l10n_in_hr_holidays/models/hr_leave.py

```diff
@@ -9,41 +12,77 @@ class HrLeave(models.Model):
     _inherit = "hr.leave"
 
     l10n_in_contains_sandwich_leaves = fields.Boolean()
+    linked_sandwich_leave_id = fields.Many2one(
+        "hr.leave",
+        string="Linked Sandwich Leave",
+        help="The leave linked to this one as part of the sandwich rule."
+    )
+    has_edge_public_leave = fields.Boolean()
+
+    def _l10n_in_apply_sandwich_rule(self, leave_days, public_holidays, employee_leaves):
+        def is_non_working_day(date):
+            return not self.resource_calendar_id._works_on_date(date) or date in holiday_dates
+
+        def count_adjacent_non_working_days(start_date, step, include_start=False):
+            current = start_date if include_start else start_date + timedelta(days=step)
```

Comment:
we think this is more clear
```suggestion
        def count_adjacent_non_working_days(start_date, reverse=False, include_start=False):
            step = -1 if reverse else 1
            current = start_date if include_start else start_date + timedelta(days=step)
```